src/cpu/aarch64/jit_aarch64_sve_512_conv_kernel.cpp

```diff
@@ -0,0 +1,4417 @@
+                    pre_offset_ker);
+        }
+    }
+    int num_zregs4ker = kw * ic_block_step;
+    int num_zregs4out = 4;
+    num_zregs4out = (28 - num_zregs4ker) / num_zregs4ker
+            ? nstl::max(num_zregs4out
+                            + ((32 - num_zregs4out - num_zregs4ker)
+                                    % (num_zregs4ker)),
+                    4)
+            : nstl::max(32 - (num_zregs4ker + ic_block_step), 4);
+    int idata_reg_offset = num_zregs4ker + num_zregs4out;
+    int num_zregs4idata = 32 - idata_reg_offset;
+
+    int pre_offset_input = -1;
+    int offset_diff_inp = -1;
+    auto load_input = [&](size_t i_offset, int zreg_idx) {
+        unsigned int IMM_MASK12 = 0xfff;
+        unsigned long long int IMM_MASK24_12 = 0xfff000;
+        unsigned int IMM_MASK24 = 0xffffff;
+
+        assert(i_offset < (1LL << 31));
+        if (ld1rw_imm_check(i_offset)) {
+            // i_offset is smaller than the maximum value of ld1rw imm
+            ld1rw(ZRegS(idata_reg_offset + (zreg_idx % num_zregs4idata)),
+                    reg_p_all_ones,
+                    ptr(reg_input, static_cast<int32_t>(i_offset)));
+
+        } else if ((pre_offset_input >= 0)
+                && ld1rw_imm_check(i_offset - pre_offset_input)) {
+            // The offset from previous access address is smaller than the maximum value of ld1rw imm
+            ld1rw(ZRegS(idata_reg_offset + (zreg_idx % num_zregs4idata)),
+                    reg_p_all_ones,
+                    ptr(reg_pre_addr_input,
+                            static_cast<int32_t>(i_offset - pre_offset_input)));
+
+        } else if ((pre_offset_input >= 0) && (offset_diff_inp >= 0)
+                && (offset_diff_inp
+                        == ((long long int)i_offset - pre_offset_input))) {
+            add(reg_pre_addr_input, reg_pre_addr_input, reg_addr_diff_input);
+            ld1rw(ZRegS(idata_reg_offset + (zreg_idx % num_zregs4idata)),
+                    reg_p_all_ones, ptr(reg_pre_addr_input));
+            pre_offset_input = i_offset;
+        } else if (ld1rw_imm_check(i_offset & IMM_MASK12)
+                && !(i_offset & ~IMM_MASK24)) {
+            // i_offset can be represented by ld1rw imm and a 12-23 bit vaule
+            add_imm(reg_pre_addr_input, reg_input, (i_offset)&IMM_MASK24_12,
+                    reg_tmp_imm);
+            ld1rw(ZRegS(idata_reg_offset + (zreg_idx % num_zregs4idata)),
+                    reg_p_all_ones,
+                    ptr(reg_pre_addr_input,
+                            static_cast<int32_t>(i_offset & IMM_MASK12)));
+            pre_offset_input = i_offset - (i_offset & IMM_MASK12);
+
+        } else if ((pre_offset_input >= 0)
+                && ld1rw_imm_check((i_offset - pre_offset_input) & IMM_MASK12)
+                && !((i_offset - pre_offset_input) & ~IMM_MASK24)) {
+            // The offset from previous access address can be represented by ld1rw imm and a 12-23 bit value
```

Comment:
I think this line might be > 80 chars.

Reply:
I don't know why "clang-format" didn't fix it, but I will fix it by hand.